### PR TITLE
chore(docs): Update analytics with gatsby-plugin-gtag

### LINKED ISSUE #27065: Update analytics doc to include google-plugin-gtag
## Summary

The existing [docs about adding Google analytics](https://www.gatsbyjs.com/docs/adding-analytics/) recommend using the `gatsby-plugin-google-analytics`. Under the hood, this plugin uses Google's `analytics.js` file.

However, Google has a [guide recommending users upgrade to `gtag.js` instead](https://developers.google.com/analytics/devguides/collection/upgrade/analyticsjs). There is another plugin ([`gatsby-plugin-gtag`](https://www.gatsbyjs.com/plugins/gatsby-plugin-gtag/)) which uses `gtag.js`.

### Motivation

The `gtag.js` file works with all Google services. Here are a few resources with more information about the difference between `gtag.js` and `analytics.js`:

- [Google FAQ about `gtag.js`](https://support.google.com/analytics/answer/7509746)
- [What’s the Difference Between analytics.js, gtag.js & ga.js? (Daan van den Bergh)](https://daan.dev/wordpress/difference-analyics-gtag-ga-js/)

## Steps to resolve this issue

- [ ] Update the [Google analytics docs](https://www.gatsbyjs.com/docs/adding-analytics/) to add a section about "Using `gatsby-plugin-gtag`".

### Open a pull request

- [ ] Open a pull request with your work including the words "closes #27065" in the pull request description

### PR DESCRIPTION
**Pull request by 98lenvi**
## Description

- Adds `gatsby-plugin-gtag` description to `analytics.md`
- Informs user on why `gtag.js` is more preferable than using `analytics.js`

## Related Issues

Fixes #27065

### CODE REVIEW

File: docs/docs/adding-analytics.md [hunk 1]
```diff
@@ -26,24 +26,24 @@ You should now have a Tracking ID; take note of it, as your website will need to
 
 You can find this tracking ID later by going to `Admin > Tracking Info > Tracking Code`.
 
-## Using `gatsby-plugin-google-analytics`
+## Using `gatsby-plugin-gtag`
 
 Now, it's time to configure Gatsby to send page views to your Google Analytics account.
 
-We are going to use `gatsby-plugin-google-analytics`. For other analytics options (including Google Analytics gtag.js and Google Tag Manager), check [other Gatsby analytics plugins](#other-gatsby-analytics-plugins).
+We are going to use `gatsby-plugin-gtag`.
```
Comment (meganesu) [inline docs/docs/adding-analytics.md:33]: I think it makes sense to leave the second sentence, so readers can easily jump to the list of other options.

```suggestion
We are going to use `gatsby-plugin-gtag`. For other analytics options (including Google Tag Manager), check [other Gatsby analytics plugins](#other-gatsby-analytics-plugins).
```

File: docs/docs/adding-analytics.md [hunk 2]
```diff
@@ -26,24 +26,24 @@ You should now have a Tracking ID; take note of it, as your website will need to
...
 
 ```shell
-npm install gatsby-plugin-google-analytics
+npm install --save gatsby-plugin-gtag
```
Comment (meganesu) [inline docs/docs/adding-analytics.md:36]: You don't need the `--save` flag here (the plugin docs should probably also be updated, but that's a separate issue).

```suggestion
npm install gatsby-plugin-gtag
```

File: docs/docs/adding-analytics.md [hunk 3]
```diff
@@ -26,24 +26,24 @@ You should now have a Tracking ID; take note of it, as your website will need to
...
 ```js:title=gatsby-config.js
 module.exports = {
   plugins: [
     {
-      resolve: `gatsby-plugin-google-analytics`,
+      resolve: `gatsby-plugin-gtag`,
       options: {
-        // replace "UA-XXXXXXXXX-X" with your own Tracking ID
-        trackingId: "UA-XXXXXXXXX-X",
+        // your google analytics tracking id
```
Comment (meganesu) [inline docs/docs/adding-analytics.md:45]: Trying to clarify further.

```suggestion
        // replace `UA-XXXXXXXXX-X` with your own Google Analytics Tracking ID
```

**Review by meganesu — CHANGES_REQUESTED** [review]
Hi @98lenvi, thanks for submitting this PR! I've left a few comments with suggestions for improvements, but other than that I think this looks good.

Thanks for your contribution! 💜

**Comment by 98lenvi** [comment]
@meganesu, Hi! Thanks for the review. I've made the changes. Please check it now 😄

**Review by meganesu — APPROVED** [review]
Looks great! ✨

Thanks for your contribution! 💜

**Comment by pfortes** [comment]
Hello, is there a reason you are using gatsby-plugin-gtag instead of gatsby-plugin-google-gtag which seams to be the oficial one, has more installs and better docs?

thank you

**Comment by meganesu** [comment]
@pfortes Oops, you're right. That's my fault. When I was writing up the initial issue, I pulled the first search result from the plugin library without realizing we had an official one. Thanks for pointing that out! 💜

I've opened a new issue for updating the doc to use the official `gatsby-plugin-google-gtag` plugin. If you're interested in making the update, feel free to comment on it and I can assign it to you.

**Comment by pfortes** [comment]
@meganesu Great, thanks for confirming this.
I will be happy to make the update, will comment on the issue.

Have a great day!

**Pull request merged**